### PR TITLE
logging: Fix initial log level in shell

### LINKED ISSUE #11866: Shell log backends started with all logs disabled
**Describe the bug**
When shell is used as log backend it starts with logging disabled for all modules. It is expected to start shell with initial log filtering set based on SHELL_BACKEND_SERIAL_LOG_LEVEL`

**To Reproduce**
Start samples/subsys/shell/shell_modules or any other sample that uses shell and type `log status` to see current log filtering settings.

**Expected behavior**
Filtering settings as configured in Kconfig.

**Impact**
Logs must be manually enabled. No option to see logs from the startup.

### PR DESCRIPTION
**Pull request by nordic-krch**
Shell log backends were initialized from shell context. After
lowering logger thread priority order has been flipped. That
revealed a bug where shell logger backend was enabled before
backend ID's has been assigned during logger initialization.

ID assignment is moved to log backend enabling function making
it independent of order of initialization.

Fixes #11866.
Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

**Comment by nordic-krch** [comment]
fyi @rlubos @jfischer-phytec-iot @jukkar Fix for logs disabled by default when shell was used.

**Review by jukkar — APPROVED** [review]
I tested this and it works ok.

**Review by rlubos — APPROVED** [review]
Works well, thanks!

**Review by jakub-uC — APPROVED** [review]
Tested and it works well

File: subsys/logging/log_core.c [hunk 1]
```diff
@@ -533,6 +505,14 @@ void log_backend_enable(struct log_backend const *const backend,
 			void *ctx,
 			u32_t level)
 {
+	u32_t id;
+
+	id = (u32_t)((u8_t *)backend - (u8_t*)log_backend_get(0))/
+			sizeof(struct log_backend);
```
Comment (jhedberg) [inline subsys/logging/log_core.c]: The way C pointer arithmetic works, a simpler way would be just `backend - log_backend_get(0)`. No division needed due to pointer types. We use the same trick e.g. here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/host/conn.c#L2254

**Review by pizi-nordic — CHANGES_REQUESTED** [review]
Please fix https://github.com/zephyrproject-rtos/zephyr/pull/11870/files#r238994607

File: subsys/logging/log_core.c (hunk 1 above)
Comment (nordic-krch) [reply subsys/logging/log_core.c]: good point. fixed.

**Review by pizi-nordic — APPROVED** [review]

**Pull request merged**